apps/src/templates/studioHomepages/BorderedCallToAction.jsx

```diff
@@ -0,0 +1,104 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import color from '@cdo/apps/util/color';
+import styleConstants from '@cdo/apps/styleConstants';
+import Button from '@cdo/apps/templates/Button';
+import {navigateToHref} from '@cdo/apps/utils';
+
+const BorderedCallToAction = ({
+  headingText,
+  descriptionText,
+  className,
+  buttonText,
+  buttonUrl,
+  buttonClass,
+  onClick,
+  solidBorder
+}) => {
+  if (!buttonUrl && !onClick) {
+    throw new Error('Expect at least one of buttonUrl / onClick');
+  }
+
+  const borderStyle = solidBorder ? styles.solidBorder : styles.dashedBorder;
+
+  return (
+    <div style={{...styles.outerBox, ...borderStyle}} className={className}>
+      <div style={styles.wordBox}>
+        <div style={styles.heading}>{headingText}</div>
+        <div style={styles.description}>{descriptionText}</div>
+      </div>
+      <Button
+        onClick={onClick || (() => navigateToHref(buttonUrl))}
+        className={buttonClass}
+        color={Button.ButtonColor.gray}
+        text={buttonText}
+        style={styles.button}
+      />
+    </div>
+  );
+};
+
+BorderedCallToAction.defaultProps = {
+  buttonColor: Button.ButtonColor.gray
+};
+
+BorderedCallToAction.propTypes = {
+  isRtl: PropTypes.bool,
```

Comment:
`isRtl` should be removed here

For my learning, could you explain how the browsers knows to rearrange the elements?

Reply:
Good catch––just updated it.

I don't know much beyond the article I linked above (https://rtlstyling.com/posts/rtl-styling/#introduction-to-rtl-styling), which explains that the "browser’s default agent" puts a `dir=rtl` attribute on the `<html>` tag at the top. When I check pages locally, I don't see this attribute ... but when I put `dir=rtl` on the tag, the direction of the site indeed changes. And when I go to code.org and change the language to a rtl language, I do see the attribute show up. This page explains more about how the attribute is determined https://html.spec.whatwg.org/multipage/dom.html#the-dir-attribute –– I'm still unclear about how it's getting set on our site.

Regardless, https://github.com/code-dot-org/code-dot-org/blob/c3325655902e82479d0a85d5adc73049810e5b66/apps/src/code-studio/isRtlRedux.js (I think written by @Hamms) explains that the DOM is the "source of truth" for setting the `isRtl`. So when we allow flexbox to rely the `dir` attribute, we are getting the same behavior that we would from `isRtl`.

Reply:
Thanks for the explanation. It is much cleaner to let the browser handle positioning, I'm wondering why we don't do this in more places and whether we should move in that direction? @madelynkasula @tim-dot-org thoughts?

Reply:
i think we should definitely move in that direction! anywhere that we can allow the browser to automatically reflow for RTL, we should. @megcrenshaw -- there is some setup you have to do locally to make this happen, but we do add `dir="rtl"` to the HTML tag in RTL languages. (FND can tell you the exact setup steps, but i know it involves being in incognito mode when you switch your language on our site, and an update to a value in your `locals.yml` file.)

i often see us using `isRtl` for edge cases where we want more granular control over styles, but i think this is legacy and an anti-pattern and will hopefully go away as we update our browser support matrix. for example, instead of adding `margin-right` to an element in LTR and `margin-left` in RTL, we should use `margin-inline-end` or a similar CSS property that will work for either flow. however, that property is relatively new and we couldn't use it with limited browser support.